src/value/parse.rs

```diff
@@ -431,6 +431,17 @@ fn single_value<I: Iterator<Item = Token>>(
     })
 }
 
+/// Parse numbers from strings *known* to contain
+/// only the characters 0-9
+fn parse_i64(s: String) -> i64 {
+    let mut output = 0i64;
+    for b in s.as_bytes() {
+        output *= 10;
+        output += i64::from(b - b'0');
+    }
+    output
+}
```

Comment:
Maybe we could use some tricks from https://kholdstare.github.io/technical/2020/05/26/faster-integer-parsing.html and use simd?

Reply:
I think SIMD might be a bit overkill for this as numbers are not usually that large. In addition, I'm not sure how well that algorithm would work for parsing integers that are not a fixed length. If `std::simd` ever becomes stable then that might be something to consider. I am wary about adding complexity in the form of `unsafe` and conditional compilation gates.